platform/core/src/redux/reducers/viewports.js

```diff
@@ -85,14 +84,24 @@ const viewports = (state = defaultState, action) => {
     // Allow fall-through
     // eslint-disable-next-line
     case SET_SPECIFIC_DATA: {
+      const layout = cloneDeep(state.layout);
+      const hasPlugin = action.data && action.data.plugin;
       const viewportIndex = useActiveViewport
         ? state.activeViewportIndex
         : action.viewportIndex;
-      currentData = cloneDeep(state.viewportSpecificData[viewportIndex]) || {};
+      const { dom } = state.viewportSpecificData[viewportIndex];
+
       viewportSpecificData = cloneDeep(state.viewportSpecificData);
-      viewportSpecificData[viewportIndex] = merge({}, currentData, action.data);
+      viewportSpecificData[viewportIndex] = {
+        dom,
+        ...action.data,
```

Comment:
I know that `action.data` is everything, but what is `action.data`?
Can we be more explicit here? Or are we SOL with the current setup?

Reply:
This is what we're calling `viewportSpecificData`, e.g.:
```javascript
{
  displaySetInstanceUid,
  seriesDate,
  seriesTime,
  seriesInstanceUid,
  seriesNumber,
  plugin,
  seriesDescription,
  modality,
  [etc...]
}
```

Reply:
Right, but we need to start destructuring "data" so it's clearer at a glance what's going into reactive state, and to make sure nothing new sneaks in unless we deliberately add it.